apps/omg_api/lib/mix/tasks/child_chain.ex

```diff
@@ -0,0 +1,45 @@
+# Copyright 2018 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule Mix.Tasks.Xomg.ChildChain.Start do
+  @moduledoc """
+    Contains mix.task to run the child chain server
+  """
+
+  use Mix.Task
+
+  @shortdoc "Start the child chain server. See Mix.Tasks.ChildChain"
+
+  def run(args) do
+    args = ensure_contains(args, "--no-start")
+    no_halt = Enum.member?(args, "--no-halt")
+    args = ensure_doesnt_contains(args, "--no-halt")
+
+    Mix.Task.run("run", args)
+    {:ok, _} = Application.ensure_all_started(:omg_api)
+    if !no_halt, do: Process.sleep(:infinity)
```

Comment:
I'll let @purbanow explain how this works, just a small note, that we probably want to rather have
```
if !no_start, do: {:ok, _} = Application.ensure_all_started...
```
or somehow handle the situation when user does `mix xomg.childchain.start --no-start`, even if it is an inconsistent invocation. Or do we not care :thinking: ?

Also mind the typo `ensure_doesnt_contain*s*`

Reply:
A) My implementation:
1) `iex -S mix xomg.child_chain.start --config ~/config.exs` ---> starts child_chain with iex
2)` mix xomg.child_chain.start --config ~/config.exs` ----> starts child_chain
3) `mix xomg.child_chain.start --config ~/config.exs  --no-start `------> same like 2)
4) `mix xomg.child_chain.start --config ~/config.exs  --no-halt` -------> finishes after run command 

B)
@pnowosie because `Mix.Task.run("run", args)` with `--no-halt` is pausing the whole run and nothing happens.

Reply:
re A) ah ok, sorry, this is getting messy, I also think I messed `--no-halt` up in our conversation, let's rethink:

`xomg...start` that starts and doesn't halt is probably the only thing that make sense.

So I think adding `--no-halt` in the command line should be a no-op - the command always keeps the application running, regarless of that flag. And of course (B) - yes, inside our task, `Task.run("run")` must have the `--no-halt` parameter, but that's a different thing, as @purbanow noted - otherwise all will freeze.

Next `xomg...start --no-start` is just `mix run --no-start`. For least astonishment, lets do the 

```
if !no_start, do: {:ok, _} = Application.ensure_all_started...
```

Reply:
@pdobacz based on you suggestions the result is supposed be following, yes ? 
1) `iex -S mix xomg.child_chain.start --config ~/config.ex`s ---> starts child_chain with iex
2) `mix xomg.child_chain.start --config ~/config.exs` ----> starts child_chain
3) `mix xomg.child_chain.start --config ~/config.exs` --no-start ------> same like 2)
4) `mix xomg.child_chain.start --config ~/config.exs` --no-halt -------> same like 2)

Reply:
yeah, that would be "--no-start ignored" option, so looks good.

Reply:
IMHO the effect of `mix xomg.child_chain.start --no-start` can be achived simpler by `mix run --no-start`, so I think we can simply ignore this flag while given...